gammapy/scripts/main.py

```diff
@@ -83,6 +83,40 @@ def cli_download(ctx, folder):
     ctx.obj = {"localfolder": folder}
 
 
+@cli.group('jupyter', short_help='Perform actions on notebooks')
+@click.option('--file', default='',
+              help='Jupyter notebook filename.')
+@click.option('--fold', default='.',
+              help='Local folder with Jupyter notebooks.')
+@click.pass_context
+def cli_jupyter(ctx, file, fold):
+    """
+    Perform a series of actions on Jupyter notebooks.
+
+    The chosen action is applied for every Jupyter notebook present in the
+    current working directory. Option --file allows to chose a single file,
+    while option --fold allows to choose a different folder to scan. These
+    options are mutually exclusive, only one is allowed.
+
+    \b
+    Examples
+    --------
+    \b
+    $ gammapy jupyter black
+    $ gammapy jupyter --file=notebook.ipynb execute
+    $ gammapy jupyter --fold=myfolder/tutorials test
+    $ gammapy jupyter stripout
+    """
+    ctx.obj = {
+        'file': file,
+        'fold': fold,
+    }
+
+    if file and fold != '.':
```

Comment:
Can't you just have one option (e.g. "src" like black, see https://github.com/ambv/black#command-line-options), and then do:
```
path = Path(src)
if path.isdir():
   paths = ...
else:
    paths = [path]
```
and the rest of the code always gets a list of paths to process?

I think that might be a bit simpler?

Reply:
If you do this you could just move the code that's currently in build_nb in called ~ 3 times into the main cli and store `paths` on the ctx, and then just do `for path in ctx.paths` for each subcommand.

This is how I would do it, but it's not very different from what you have, just a matter of taste. Do as you prefer.